executor/src/main/java/com/redhat/service/smartevents/executor/ExecutorImpl.java

```diff
@@ -70,25 +69,14 @@ public void onEvent(CloudEvent event, Map<String, String> traceHeaders) {
     private void process(CloudEvent event, Map<String, String> traceHeaders) {
         Map<String, Object> eventMap = toEventMap(event);
 
-        LOG.debug("Received event with id '{}' and type '{}' in processor with name '{}' of bridge '{}",
-                event.getId(),
-                event.getType(),
-                processor.getName(),
-                processor.getBridgeId());
+        LOG.debug("Received event with id '{}' and type '{}' in processor with name '{}' of bridge '{}", event.getId(), event.getType(), processor.getName(), processor.getBridgeId());
 
         // Filter evaluation
         if (!matchesFilters(eventMap)) {
-            String message = String.format("Filters of processor '%s' did not match for event with id '%s' and type '%s'",
-                    processor.getId(),
-                    event.getId(),
-                    event.getType());
-            LOG.debug(message);
-            throw new EventRemovedByProcessorFilterException(message);
+            LOG.debug("Filters of processor '{}' did not match for event with id '{}' and type '{}'", processor.getId(), event.getId(), event.getType());
```

Comment:
The customer wants notification of events filtered out by RHOSE. This exception would cause the event to be written to the back channel\DLQ. You will want to ensure such a mechanism still exists.

Reply:
We discussed about a possible "event lost" error if none of the filters have matched and the event is discharged but if we do the implementation at executor/processor level it generates an error for every processor that doesn't match.
Given that NS expects to have only 1 processor to match, this is going to produce (N-1) errors for every event (with N = number of users using the service).
To implement the "event lost" we need to find some different mechanism (probably Knative trigger API can help for this).

Reply:
This is why I have kept going on about the "reduce" feature we need to provide if we are not exposing every processor/event/status update to NS. That cannot be ran as a Processor and hence the Sidecar was a better option. The DLQ is meant to be an internal implementation to capture every processor/event/status change. The reduction mechanism was to consolidate/reconcile before emitting a single update to Users. For NS a Processor is good enough. For more complex users a sidecar was @r00ta and my preference.

Reply:
my2c: if the user wants to know if an event did not match a filter he/she might be interested in **logs** and not in **errors**. As I would not define that situation as an `error`, given that it would work **as expected**.
Some examples of logs are: 
1) the event did not match the filters
2) the event with id `x` was processed at time `z`
Some examples of errors are: 
1) We fail to send the Webhook for whatever reason and the event could not be delivered
2) The event is not processed within a reasonable amount of time (TBD). For example the event stays in kafka for more than X minutes without being consumed for whatever reason
3) The event does not contain a key that is required by the qute template -> we could not process the event
and so on..

Reply:
The DLQ is a misdemeanor. It is just a back channel topic to capture informational status updates for events. The DLQ was proposed as an "out of the box" solution for error capture but was also needed for information messages. My initial implementation did not use a DLQ but a more programmatic use of what was an explicit back channel topic. This whole requirement has a lot of overlap with distributed tracing, something I have tried to communicate.

Reply:
> This whole requirement has a lot of overlap with distributed tracing, something I have tried to communicate.

I completely agree, this is just the first step to start addressing the "black box" nature of current implementation but it is not enough for many use cases where users want to have a more general "insight" of the status of the system. This initial PR aims to provide an error handling API/pipeline following the common pattern of DLQ. 

For example the idea of "reduce" to aggregate messages (errors or even just tracing information) so that part of this complexity is hidden to the user can make totally sense as enhancement.